src/style/shared/number-input.less

```diff
@@ -32,10 +32,6 @@
     border-bottom: @hairline solid @item-active;
 }
 
-.number-input:disabled:hover {
```

Comment:
It looked bad. I'll alert @placegraphichere in case he wants to change the styling more, but without this it seems fine.